python/cudf_polars/cudf_polars/dsl/expressions/rolling.py

```diff
@@ -210,9 +260,341 @@ def __init__(
             for ne in self.named_aggs
             for v in (ne.value,)
             if isinstance(v, expr.Agg)
+            or (
+                isinstance(v, expr.UnaryFunction)
+                and v.name in {"rank", "fill_null_with_strategy", "cum_sum"}
+            )
         ]
         self.by_count = len(by_expr)
-        self.children = tuple(by_expr) + tuple(child_deps)
+        self.children = tuple(by_expr) + tuple(self._order_by_exprs) + tuple(child_deps)
+
+    @singledispatchmethod
+    def _apply_unary_op(
+        self,
+        op: UnaryOp,
+        df: DataFrame,
+        grouper: plc.groupby.GroupBy,
+    ) -> tuple[list[str], list[DataType], list[plc.Table]]:
+        raise NotImplementedError(
+            f"Unsupported unary op: {type(op).__name__}"
+        )  # pragma: no cover; translation raises first
+
+    @_apply_unary_op.register
+    def _(
+        self,
+        op: RankOp,
+        df: DataFrame,
+        grouper: plc.groupby.GroupBy,
+    ) -> tuple[list[str], list[DataType], list[plc.Table]]:
+        rank_named = op.named_exprs
+        order_index = op.order_index
+        by_cols_for_scan = op.by_cols_for_scan
+
+        rank_requests: list[plc.groupby.GroupByRequest] = []
+        rank_out_names: list[str] = []
+        rank_out_dtypes: list[DataType] = []
+
+        for ne in rank_named:
+            rank_expr = ne.value
+            (child_expr,) = rank_expr.children
+            val_col = child_expr.evaluate(df, context=ExecutionContext.FRAME).obj
+            if order_index is not None:
+                val_col = plc.copying.gather(
+                    plc.Table([val_col]),
+                    order_index,
+                    plc.copying.OutOfBoundsPolicy.NULLIFY,
+                ).columns()[0]
```

Comment:
Should we consider accumulating a `plc.Table` of all the `val_col`s and calling `plc.copying.gather` once since the `order_index` is used for all these columns?